addressing_address_map.go

```diff
@@ -0,0 +1,202 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+// AddressMap contains information about an address map.
+type AddressMap struct {
+	ID           string                 `json:"id"`
+	Description  string                 `json:"description"`
+	DefaultSNI   string                 `json:"default_sni"`
+	Enabled      bool                   `json:"enabled"`
+	CanDelete    bool                   `json:"can_delete"`
+	CanModifyIPs bool                   `json:"can_modify_ips"`
+	Memberships  []AddressMapMembership `json:"memberships"`
+	IPs          []AddressMapIP         `json:"ips"`
+	CreatedAt    *time.Time             `json:"created_at"`
+	ModifiedAt   *time.Time             `json:"modified_at"`
+}
+
+type AddressMapIP struct {
+	IP        string     `json:"ip"`
+	CreatedAt *time.Time `json:"created_at"`
+}
+
+type AddressMapMembership struct {
+	Identifier string                   `json:"identifier"`
+	CanDelete  bool                     `json:"can_delete"`
+	Kind       AddressMapMembershipKind `json:"kind"`
+	CreatedAt  *time.Time               `json:"created_at"`
+}
+
+type AddressMapMembershipKind string
+
+const (
+	AddressMapMembershipZone    AddressMapMembershipKind = "zone"
+	AddressMapMembershipAccount AddressMapMembershipKind = "account"
+)
+
+// ListAddressMapResponse contains a slice of address maps.
+type ListAddressMapResponse struct {
+	Response
+	Result []AddressMap `json:"result"`
+}
+
+// GetAddressMapResponse contains a specific address map's API Response.
+type GetAddressMapResponse struct {
+	Response
+	Result AddressMap `json:"result"`
+}
+
+// CreateAddressMapRequest contains basic information about address map to be created.
+type CreateAddressMapRequest struct {
+	Description string `json:"description"`
+	Enabled     bool   `json:"enabled"`
+}
+
+// AddressMapPatch contains information about address map for updates.
+type AddressMapPatch struct {
+	Description *string `json:"description,omitempty"`
+	Enabled     *bool   `json:"enabled,omitempty"`
+	DefaultSNI  *string `json:"default_sni,omitempty"`
+}
+
+// ListAddressMaps lists all address maps owned by the account.
+//
+// API reference: https://developers.cloudflare.com/api/operations/ip-address-management-address-maps-list-address-maps
+func (api *API) ListAddressMaps(ctx context.Context, accountID string) ([]AddressMap, error) {
```

Comment:
i've outlined some conventions for new functionality at https://github.com/cloudflare/cloudflare-go/blob/master/docs/experimental.md. in essence:

- all methods have max 3 parameters
- the third parameter is always a struct (or a single ID for `Get` methods)
- the second parameter is a `ResourceContainer`
- where possible, we stick to `Get`, `List`, `Update`, `Create` and `Delete` method prefixes for consistency
- `List` operations automatically paginate all results

you can check out https://github.com/cloudflare/cloudflare-go/blob/master/tunnel.go for prior art if you're wondering how it all hangs together.

Reply:
I've made some changes according to these experimental items. The list api itself in this case does not support paging. Do I need to do anything with it?

Reply:
as long as the signature matches, that's fine. we have plenty of APIs that don't paginate but need to look the same.